core/src/main/scala/kafka/utils/Pool.scala

```diff
@@ -57,21 +56,10 @@ class Pool[K,V](valueFactory: Option[K => V] = None) extends Iterable[(K, V)] {
     * @param createValue Factory function.
     * @return The final value associated with the key.
     */
-  def getAndMaybePut(key: K, createValue: => V): V = {
-    val current = pool.get(key)
-    if (current == null) {
-      createLock synchronized {
-        val current = pool.get(key)
-        if (current == null) {
-          val value = createValue
-          pool.put(key, value)
-          value
-        }
-        else current
-      }
-    }
-    else current
-  }
+  def getAndMaybePut(key: K, createValue: => V): V =
+    pool.computeIfAbsent(key, new java.util.function.Function[K,V] {
```

Comment:
Use a lambda please :)

Reply:
Oh, this is Scala code and lambdas won't work in Scala 2.11 (but will in Scala 2.12).

Reply:
ok, I'll leave it as-is then